rm_manual:
  robot_type: "standard"
  chassis:
    topic: "/cmd_chassis"
    accel_x: 10.0
    accel_y: 10.0
    accel_z: 20.0
    safety_power: 50.0
    capacitor_threshold: 0.1
    charge_power: 20
    extra_power: 30
    burst_power: 220
  vel:
    topic: "/cmd_vel"
    max_linear_x: 4.0
    max_linear_y: 4.0
    max_angular_z: 10.0
  gimbal:
    topic: "/cmd_gimbal"
    max_yaw_vel: 12.56
    max_pitch_vel: 8.37
    track_timeout: 0.5
    target_cost_function:
      k_pos: 1.0
      k_vel: 1.0
      k_hp: 1.0
      k_freq: 1.0
      timeout: 0.05
  shooter:
    topic: "/cmd_shoot"
    gimbal_error_limit: 0.05
    heat_limit:
      expect_shoot_frequency_1: 5
      expect_shoot_frequency_2: 8
      expect_shoot_frequency_3: 12
      burst_shoot_frequency: 20
      safe_shoot_frequency: 5
      heat_coeff: 8.0
      type: "ID1_17MM"
  cover:
    topic: "/controllers/cover_controller/command"
    open_pos: -2.0
    close_pos: 0.0
  enemy_color_switch:
    error_amount_limit: 10
  controllers_list:
    - controllers/robot_state_controller
    - controllers/joint_state_controller
    - controllers/chassis_controller
    - controllers/gimbal_controller
    - controllers/shooter_controller
    - controllers/cover_controller
    - controllers/trigger_calibration_controller
    - controllers/cover_calibration_controller
  state_controllers_switch:
    start_controllers:
      - controllers/robot_state_controller
      - controllers/joint_state_controller
  base_controllers_switch:
    start_controllers:
      - controllers/chassis_controller
      - controllers/gimbal_controller
      - controllers/shooter_controller
      - controllers/cover_controller
  calibration_manager:
    - step: cover
      switch:
        start_controllers:
          - controllers/cover_calibration_controller
        stop_controllers:
          - controllers/cover_controller
      query:
        service_name: /controllers/cover_calibration_controller/is_calibrated
    - step: trigger
      switch:
        start_controllers:
          - controllers/trigger_calibration_controller
        stop_controllers:
          - controllers/shooter_controller
      query:
        service_name: /controllers/trigger_calibration_controller/is_calibrated

controllers:
  cover_controller:
    type: effort_controllers/JointPositionController
    joint: "cover_joint"
    pid: { p: 15.0, i: 0.0, d: 1.2, i_clamp_max: 0.0, i_clamp_min: 0.0, antiwindup: true, publish_state: true }
